code/modules/mob/living/bot/farmbot.dm

```diff
@@ -129,11 +129,12 @@
 			if(path.len && frustration < 5)
 				if(path[1] == loc)
 					path -= path[1]
-				var/t = step_towards(src, path[1])
-				if(t)
-					path -= path[1]
-				else
-					++frustration
+				if(path.len)
```

Comment:
`path.len` is already checked up here.

Reply:
yes but it loses the first reference , and if the only reference was that one ,then it errors.